pkg/ccl/sqlproxyccl/idle_disconnect_connection.go

```diff
@@ -0,0 +1,61 @@
+// Copyright 2020 The Cockroach Authors.
+//
+// Licensed as a CockroachDB Enterprise file under the Cockroach Community
+// License (the "License"); you may not use this file except in compliance with
+// the License. You may obtain a copy of the License at
+//
+//     https://github.com/cockroachdb/cockroach/blob/master/licenses/CCL.txt
+
+package sqlproxyccl
+
+import (
+	"net"
+	"time"
+
+	"github.com/cockroachdb/cockroach/pkg/util/timeutil"
+	"github.com/cockroachdb/errors"
+)
+
+// IdleDisconnectConnection is a wrapper around net.Conn that disconnects if connection is idle
+type IdleDisconnectConnection struct {
+	net.Conn
+	timeout time.Duration
+}
+
+var notSupportedError = errors.Errorf("Not supported for IdleDisconnectConnection")
+
+// Read reads data from the connection with timeout.
+func (c IdleDisconnectConnection) Read(b []byte) (n int, err error) {
+	if err := c.Conn.SetReadDeadline(timeutil.Now().Add(c.timeout)); err != nil {
+		return 0, err
+	}
+	return c.Conn.Read(b)
+}
+
+// Write writes data to the connection and sets the read timeout.
+func (c IdleDisconnectConnection) Write(b []byte) (n int, err error) {
+	if err := c.Conn.SetReadDeadline(timeutil.Now().Add(c.timeout)); err != nil {
```

Comment:
Using a read deadline is not obvious, add a comment perhaps?

Reply:
added.